### PR TITLE
[IMP] project: merge date_deadline and planned_date_end

### PR DESCRIPTION
**Pull request by RaoufGhrissi**
**[IMP] project: merge date_deadline and planned_date_end**
**Before this commit:**
- The existing of two dates behaving as and end date for project_task
(date_deadline and planned_date_end) is confusing.

**In this commit:**
- The main goal is to simplify the interface by having one field/widget
that acts as both the deadline and the end date, instead of having two separate fields.
So we removed planned_date_end (changes can be seen in the enterprise related pr (link at
the end) and changed the date_deadline type from date to datetime in project_task and
report_project_task_user

Related PRs:
Enterprise: https://github.com/odoo/enterprise/pull/40866

task-3084978

### CODE REVIEW

**Review by auon-odoo — APPROVED** [review]
lgtm :zap:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks for your work :slightly_smiling_face:

File: addons/project/views/project_sharing_project_task_views.xml [hunk 1]
```diff
@@ -173,7 +173,7 @@
                             <field name="company_id" invisible="1" />
                             <field name="state" invisible="1" />
                             <field name="partner_id" options="{'no_open': True, 'no_create': True, 'no_edit': True}" invisible="not project_id"/>
-                            <field name="date_deadline" invisible="state in ['1_done', '1_canceled']"/>
+                            <field name="date_deadline" invisible="state in ['1_done', '1_canceled']" decoration-danger="date_deadline &lt; current_date"/>
```
Comment (auon-odoo) [inline addons/project/views/project_sharing_project_task_views.xml:176]: shouldn't it be if the deadline is after today?

Comment (RaoufGhrissi) [reply addons/project/views/project_sharing_project_task_views.xml:176]: it's red when we exceed the deadline

Comment (auon-odoo) [reply addons/project/views/project_sharing_project_task_views.xml:176]: euh yes indeed. me dumdumb.

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks again for your work. 🙂 

robodoo r+

**Comment by xavierbol** [comment]
@robodoo r+

**Pull request closed without merging**